### PR TITLE
Druid v0.6 compatibility (second try)

### PR DESCRIPTION
**Pull request by dimaslv**
These commits allowed me to set up latest superset with our company's Druid v0.6 cluster. I have tried not to break anything for any newer versions.
This is second try for those changes to merge into upstream, first pull request was here https://github.com/apache/incubator-superset/pull/2331.

### CODE REVIEW

**Comment by xrmx** [comment]
Just a couple of thoughts:
- any chance that pydruid sqlalchemy dialect already handles that?
- passing the version as parameter on an object does not look that nice, either storing the version as an attribute in the object or creating a subclass for older apis may look better

File: superset/connectors/druid/models.py [hunk 1]
```diff
@@ -207,13 +211,14 @@ def refresh(self, datasource_names, merge_flag, refreshAll):
                     if datatype == 'hyperUnique' or datatype == 'thetaSketch':
                         col_obj.count_distinct = True
                     # Allow sum/min/max for long or double
-                    if datatype == 'LONG' or datatype == 'DOUBLE':
+                    if datatype == 'LONG' or datatype == 'DOUBLE' or datatype == 'FLOAT':
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: let's use `datetype in ('foo', 'bar')`

File: superset/connectors/druid/models.py [hunk 2]
```diff
@@ -119,7 +119,11 @@ def get_druid_version(self):
         endpoint = (
             'http://{obj.coordinator_host}:{obj.coordinator_port}/status'
         ).format(obj=self)
-        return json.loads(requests.get(endpoint).text)['version']
+        ver = json.loads(requests.get(endpoint).text)['version']
+        if ver is None:
```
Comment (mistercrunch) [inline superset/connectors/druid/models.py]: `return ver or "0"` instead of the 4 lines

**Comment by mistercrunch** [comment]
@xrmx doesn't have SQL support at all. I think SQLAlchemy access to druid might require `0.10` or `0.11`+ Druid backend

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Can we just reference `self.cluster.druid_version` instead of passing it around?

**Pull request closed without merging**